lib/sycamore/sycamore/transforms/extract_document_structure.py

```diff
@@ -16,6 +18,50 @@ def extract(document: Document) -> Union[Document, HierarchicalDocument]:
         pass
 
 
+class StructureByImages(DocumentStructure):
+    @staticmethod
+    def extract(doc: Document) -> HierarchicalDocument:
+        doc = HierarchicalDocument(doc.data)
+        doc.data["relationships"] = doc.get("relationships", {})
+        doc.data["label"] = doc.get("label", "DOCUMENT")
+
+        images = [child for child in doc.children if child.type == "Image"]
+        doc.children = []
+        for image in images:
+            initial_page = HierarchicalDocument(
+                {
+                    "type": "Section-header",
+                    "bbox": (0, 0, 0, 0),
+                    "properties": {"score": 1, "page_number": 1},
+                    "text_representation": "Document",
+                    "binary_representation": b"Front Page",
+                    "relationships": {},
+                    "label": "SECTION",
+                }
+            )
+            rel_1 = {
+                "TYPE": "SECTION_OF",
+                "properties": {},
+                "START_ID": initial_page.doc_id,
+                "START_LABEL": "SECTION",
+                "END_ID": doc.doc_id,
+                "END_LABEL": "DOCUMENT",
+            }
+            rel_2 = {
+                "TYPE": "PART_OF",
+                "properties": {},
+                "START_ID": image.doc_id,
+                "START_LABEL": "IMAGE",
+                "END_ID": initial_page.doc_id,
+                "END_LABEL": "SECTION",
+            }
+            doc.data["relationships"][str(uuid.uuid4())] = rel_1
+            initial_page.data["relationships"][str(uuid.uuid4())] = rel_2
+            initial_page.children.append(image)
+            doc.children.append(initial_page)
```

Comment:
so just to get this straight:
the new InitialPage doc is a SECTIONOF the document
the image is a PARTOF the new initialpage

Reply:
yes, thats right